### ISSUE TITLE
Generated macros for npm package bin should trigger `npm_package_bin` if `stdout` is used

### ISSUE THREAD

**Issue by jfirebaugh** [bazelbuild/rules_nodejs#3403]
# 🚀 feature request

### Relevant Rules

Any [generated macro for npm packages with `bin` entries](https://docs.aspect.build/bazelbuild/rules_nodejs/5.3.0/docs/repositories.html#generated-macros-for-npm-packages-with-bin-entries).

### Description

I am using [emscripten-library-generator](https://www.npmjs.com/package/emscripten-library-generator), which has a `bin` that writes to stdout (and no option to write somewhere else). I would like to write:

```
load("@npm//emscripten-library-generator:index.bzl", "emscripten_library_generator")

emscripten_library_generator(
    name = "gen",
    ...
    stdout = "library.js",
)
```

This currently triggers:

```
ERROR: [snip path]: //:gen: no such attribute 'stdout' in 'nodejs_binary' rule
```

### Describe the solution you'd like

If one of the generated macros is used with [the `stdout` attribute](https://github.com/bazelbuild/rules_nodejs/pull/1940), it should use `npm_package_bin` rather than `nodejs_binary`.

### Describe alternatives you've considered

Revert to using `genrule`.

**Comment by Whoaa512** [comment]
I believe this works, at least it does in 5.4.0
